config/file_config.go

```diff
@@ -306,6 +306,7 @@ type CollectionConfig struct {
 	MaxAlloc              MemorySize `yaml:"MaxAlloc"`
 	DisableRedistribution bool       `yaml:"DisableRedistribution"`
 	ShutdownDelay         Duration   `yaml:"ShutdownDelay" default:"15s"`
+	ForceTraceLocality    bool       `yaml:"ForceTraceLocality"`
```

Comment:
I'm not sure `Force` feels right here, maybe we could name it `EnableTraceLocality` instead? Also, should we provide the default of `false` here too for clarity?

Also, all of our references to this setting have to invert the value (eg !i.Config.GetCollectionConfig().ForceTraceLocality) - we could invert the config option and default true.

eg `DisableTraceLocality` with default true

Reply:
I like the name `EnableTraceLocality`. However, defaulting to true adds complexity when we access this field due to nil pointer issue. An example use case is the `AddSpanCountToRoot`. We had to do this for backwards compatibility. I would prefer not have to continue the pattern
```
AddSpanCountToRoot     *DefaultTrue `yaml:"AddSpanCountToRoot" default:"true"` // Avoid pointer woe on access, use GetAddSpanCountToRoot() instead.
```